src/go/cmd/http-relay-server/server/broker.go

```diff
@@ -235,12 +237,21 @@ func (r *broker) GetRequestStream(id string) ([]byte, bool) {
 func (r *broker) PutRequestStream(id string, data []byte) bool {
 	r.m.Lock()
 	pr := r.resp[id]
+	pr.requestStreamMutex.Lock()
```

Comment:
I think this needs to be after the pr == nil check. If so you'll need to bring the nil check inside the critical section and make sure r.m is unlocked on both paths similar to SendResponse().

Reply:
43fa715eb89c805f31db4ec610fea574e16ef732